src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .map((styles) => styles.trim())
+    .filter((styles) => styles !== '')
+    .reduce((cssStyles, styles) => {
+      const [key, value] = styles.split(':').map((item) => item.trim());
+
+      if (key && value) {
+        cssStyles[key] = value;
+      }
+
+      return cssStyles;
```

Comment:
[CODE KNOWLEDGE] - Don't use iteration methods to modify an already existing array/object. The `reduce` method is being used correctly here to produce a new object from the array of styles.